File: src/main/java/com/trade/securities/application/service/FscStockPriceService.java
package com.trade.securities.application.service;

import com.trade.securities.application.port.in.GetFscStockPriceUseCase;
import com.trade.securities.application.port.out.LoadFscStockPricePort;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Map;

@Service
@RequiredArgsConstructor
public class FscStockPriceService implements GetFscStockPriceUseCase {

    private final LoadFscStockPricePort loadFscStockPricePort;

    @Override
    public Map<String, Object> getStockPrices(int pageNo, int numOfRows, String basDt, String itmsNm,
            String likeItmsNm) {
        return loadFscStockPricePort.loadStockPrices(pageNo, numOfRows, basDt, itmsNm, likeItmsNm);
    }
}

File: src/main/kotlin/com/trade/securities/application/service/FscStockPriceService.kt
package com.trade.securities.application.service

import com.trade.securities.application.port.`in`.GetFscStockPriceUseCase
import com.trade.securities.application.port.out.LoadFscStockPricePort
import org.springframework.stereotype.Service

@Service
class FscStockPriceService(
    private val loadFscStockPricePort: LoadFscStockPricePort
) : GetFscStockPriceUseCase {

    override fun getStockPrices(pageNo: Int, numOfRows: Int, basDt: String, itmsNm: String?, likeItmsNm: String?): Map<String, Any?> {
        return loadFscStockPricePort.loadStockPrices(pageNo, numOfRows, basDt, itmsNm, likeItmsNm)
    }
}
